### DISCUSSION TITLE
parser: 关于ImplicitCastExpr的问题

### DISCUSSION THREAD

**Discussion by ma-h-m** [arcsysu/SYsU-lang#87] [Q&A]
在样例15_array_test3.sysu.c中，main函数部分出现如下部分。请问这里的func(a)调用出现的第二个ImplicitCastExpr该如何判断？

或者说，我该如何判断什么时候会出现ImplicitCastExpr?

`int main(){

    putint(func(a));

    return 0;
}`

 -FunctionDecl 0x191e5f0 <line:18:1, line:21:1> line:18:5 main 'int ()'
  -CompoundStmt 0x191e850 <col:11, line:21:1>
    |-CallExpr 0x191e7f8 <line:19:5, col:25> 'void'
    | |-ImplicitCastExpr 0x191e7e0 <col:5> 'void (*)(int)' <FunctionToPointerDecay>
    | | -DeclRefExpr 0x191e690 <col:5> 'void (int)' Function 0x18fd5f8 '_sysy_putint' 'void (int)'
    | -CallExpr 0x191e768 <col:18, col:24> 'int'
    |   |-ImplicitCastExpr 0x191e750 <col:18> 'int (*)(int (*)[5])' <FunctionToPointerDecay>
    |   | -DeclRefExpr 0x191e6b0 <col:18> 'int (int (*)[5])' Function 0x191dde8 'func' 'int (int (*)[5])'
    |   -ImplicitCastExpr 0x191e790 <col:23> 'int (*)[5]' <ArrayToPointerDecay>
    |     -DeclRefExpr 0x191e6d0 <col:23> 'int [5][5]' lvalue Var 0x18fdcc8 'a' 'int [5][5]'
    -ReturnStmt 0x191e840 <line:20:5, col:12>
      -IntegerLiteral 0x191e820 <col:12> 'int' 0

**Comment by wufeng15226** [comment]
你具体指的是哪一个 ImplicitCastExpr？`clang -cc1 -ast-dump` 的输出中有隐式类型转换的类型，可以在一定程度上帮助你理解。
